nixos/modules/services/misc/taskserver/doc.xml

```diff
@@ -0,0 +1,144 @@
+      <quote>server certificates</quote>.
+    </para>
+
+    <para>
+      So in order to make it easier to handle your own CA, there is a helper
+      tool called <command>nixos-taskserver</command> which manages the custom
+      CA along with Taskserver organisations, users and groups.
+    </para>
+
+    <para>
+      While the client certificates in Taskserver only authenticate whether a
+      user is allowed to connect, every user has its own UUID which identifies
+      it as an entity.
+    </para>
+
+    <para>
+      With <command>nixos-taskserver</command> the client certificate is created
+      along with the UUID of the user, so it handles all of the credentials
+      needed in order to setup the Taskwarrior client to work with a Taskserver.
+    </para>
+  </section>
+
+  <section>
+    <title>The nixos-taskserver tool</title>
+
+    <para>
+      Because Taskserver by default only provides scripts to setup users
+      imperatively, the <command>nixos-taskserver</command> tool is used for
+      addition and deletion of organisations along with users and groups defined
+      by <option>services.taskserver.organisations</option> and as well for
+      imperative set up.
+    </para>
+
+    <para>
+      The tool is designed to not interfere if the command is used to manually
+      set up some organisations, users or groups.
+    </para>
+
+    <para>
+      For example if you add a new organisation using
+      <command>nixos-taskserver org add foo</command>, the organisation is not
+      modified and deleted no matter what you define in
+      <option>services.taskserver.organisations</option>, even if you're adding
+      the same organisation in that option.
+    </para>
+
+    <para>
+      The tool is modelled to imitate the official <command>taskd</command>
+      command, documentation for each subcommand can be shown by using the
+      <option>--help</option> switch.
+    </para>
+  </section>
+  <section>
+    <title>Declarative/automatic CA management</title>
+
+    <para>
+      Everything is done according to what you specify in the module options,
+      however in order to set up a Taskwarrior client for synchronisation with a
+      Taskserver instance, you have to transfer the keys and certificates to the
+      client machine.
+    </para>
+
+    <para>
+      This is done using
+      <command>nixos-taskserver user export $orgname $username</command> which
+      is printing a shell script fragment to stdout which can either be used
+      verbatim or adjusted to import the user on the client machine.
+    </para>
+
+    <para>
+      For example, let's say you have the following configuration:
+<screen>
+{
+  services.taskserver.enable = true;
+  services.taskserver.fqdn = "server";
+  services.taskserver.listenHost = "::";
+  services.taskserver.organisations.NixOS.users = [ "alice" ];
+}
+</screen>
+      This creates an organisation called <literal>NixOS</literal> with the user
```

Comment:
Choose a better organization name in this example, such that it is less confusing.  Maybe something like `my-company`